docs/DeveloperGuide.md

```diff
@@ -161,10 +163,56 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 --------------------------------------------------------------------------------------------------------------------
 
-### **Implementation**
-
+# **Implementation**
 This section describes some noteworthy details on how certain features are implemented.
 
+## Optional Fields
+
+Certain fields in the Internship class are made optional to offer flexibility to the users.
+This report outlines the technical details of the implementation, including the different states, usage scenarios, design considerations, alternatives, and their implications.
+
+### Implementation Details (How Is It Implemented):
+
+**Field Conversion:** <br>
+Fields such as location and role have been converted to optional types (where default values have been set should the user not input any values for them).
+This provides flexibility in representing internship data.
+
+**Constructor Modification:** <br>
+The constructor for the Internship class consists of 2 optional fields (`role` and `location`), ensuring compatibility with the new optional field structure.
+The add command parser sets default values for the 2 aforementioned optional fields, should the user not input any values.
+
+**Equality Comparison:** <br>
+The equals() and isSameInternship() methods account for optional fields differently based on the scenario:
+1. If two internships have identical /company, /poc, /desc fields and have the same /role and /loc, they will be considered as identical internships.
```

Comment:
Is this contact email and contact phone also needed to check for equality?

Reply:
oh mb. i shd have used poc fields instead of /poc